x-pack/plugins/security/public/management/api_keys/api_keys_grid/api_key_flyout.tsx

```diff
@@ -31,119 +32,119 @@ import { i18n } from '@kbn/i18n';
 import { FormattedMessage } from '@kbn/i18n-react';
 import { CodeEditorField, useKibana } from '@kbn/kibana-react-plugin/public';
 
-import type { ApiKey, ApiKeyRoleDescriptors, AuthenticatedUser } from '../../../../common/model';
+import type { ApiKeyRoleDescriptors } from '../../../../common/model';
 import { DocLink } from '../../../components/doc_link';
+import { FormField } from '../../../components/form_field';
 import type { FormFlyoutProps } from '../../../components/form_flyout';
 import { FormFlyout } from '../../../components/form_flyout';
+import { FormRow } from '../../../components/form_row';
 import { useCurrentUser } from '../../../components/use_current_user';
-import { useForm } from '../../../components/use_form';
-import type { ValidationErrors } from '../../../components/use_form';
 import { useInitialFocus } from '../../../components/use_initial_focus';
 import { RolesAPIClient } from '../../roles/roles_api_client';
 import { APIKeysAPIClient } from '../api_keys_api_client';
 import type {
-  CreateApiKeyRequest,
-  CreateApiKeyResponse,
-  UpdateApiKeyRequest,
-  UpdateApiKeyResponse,
+  CreateAPIKeyParams,
+  CreateAPIKeyResult,
+  UpdateAPIKeyParams,
+  UpdateAPIKeyResult,
 } from '../api_keys_api_client';
+import type { CategorizedApiKey } from './api_keys_grid_page';
+import { ApiKeyBadge, ApiKeyStatus, TimeToolTip, UsernameWithIcon } from './api_keys_grid_page';
 
 export interface ApiKeyFormValues {
   name: string;
+  type: string;
   expiration: string;
   customExpiration: boolean;
   customPrivileges: boolean;
   includeMetadata: boolean;
+  access: string;
   role_descriptors: string;
   metadata: string;
 }
 
-export interface ApiKeyFlyoutProps {
+interface CommonApiKeyFlyoutProps {
   defaultValues?: ApiKeyFormValues;
-  onSuccess?: (
-    createApiKeyResponse: CreateApiKeyResponse | undefined,
-    updateApiKeyResponse: UpdateApiKeyResponse | undefined
-  ) => void;
   onCancel: FormFlyoutProps['onCancel'];
-  apiKey?: ApiKey;
-  readonly?: boolean;
+  canManageCrossClusterApiKeys?: boolean;
+  readOnly?: boolean;
 }
 
+interface CreateApiKeyFlyoutProps extends CommonApiKeyFlyoutProps {
+  onSuccess?: (createApiKeyResponse: CreateAPIKeyResult) => void;
+}
+
+interface UpdateApiKeyFlyoutProps extends CommonApiKeyFlyoutProps {
+  onSuccess?: (updateApiKeyResponse: UpdateAPIKeyResult) => void;
+  apiKey: CategorizedApiKey;
+}
+
+export type ApiKeyFlyoutProps = ExclusiveUnion<CreateApiKeyFlyoutProps, UpdateApiKeyFlyoutProps>;
+
 const defaultDefaultValues: ApiKeyFormValues = {
```

Comment:
Is this an old typo? Or is it really the default default values?

Reply:
This was not a typo since these were the defaults for the `defaultValues` prop. I agree that it's a bit clunky though so have renamed `defaultValues` to `initialValues`.